google/datalab/pipeline/_pipeline.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+from google.datalab import utils
+from enum import Enum
+
+class Operator(Enum):
+  BigQuery = 'bq'
+  BigQueryTableDelete = 'bq-table-delete'
+  BigQueryToBigQuery = 'bq-to-bq'
+  BigQueryToCloudStorage = 'bq-to-gcs'
+  Bash = 'bash'
+
+class Pipeline(object):
+
+  _imports = """
+from airflow import DAG
+from airflow.operators.bash_operator import BashOperator
+from airflow.contrib.operators.bigquery_operator import BigQueryOperator
+from airflow.contrib.operators.bigquery_table_delete_operator import BigQueryTableDeleteOperator
+from airflow.contrib.operators.bigquery_to_bigquery import BigQueryToBigQueryOperator
+from airflow.contrib.operators.bigquery_to_gcs import BigQueryToCloudStorageOperator
+from datetime import datetime, timedelta
+
+"""
+
+  _default_args_format = """
+    'owner': 'Datalab',
+    'depends_on_past': False,
+    'email': ['{0}'],
+    'start_date': datetime.strptime('{1}', '{3}'),
+    'end_date': datetime.strptime('{2}', '{3}'),
+    'email_on_failure': True,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': timedelta(minutes=1),
+"""
+
+  def __init__(self, spec_str, name, env=None):
+    self._spec_str = spec_str
+    self._env = env or {}
+    self._name = name
+
+  @property
+  def spec(self):
+    return self._spec_str
+
+  @property
+  def py(self):
```

Comment:
Add a Docstring for this function at a minimum, since it's the main public one.

Docstrings for all the rest wouldn't hurt.